wire-ios-transport/Source/Public/BackendInfo.swift

```diff
@@ -29,44 +29,85 @@ public enum BackendInfo {
+
+    /// The `UserDefaults` used to store backend configuration info.
+    ///
+    /// - Note: By default this is `UserDefaults.standard`. However, this property is currently overwritten by the main
+    /// app on startup.
+
     public static var storage = UserDefaults.standard
 
     /// The currently selected API Version.
 
     public static var apiVersion: APIVersion? {
         get { return apiVersion(for: Key.selectedAPIVersion) }
-        set { storage.set(newValue?.rawValue, forKey: Key.selectedAPIVersion.rawValue) }
+        set { _storage.set(newValue?.rawValue, forKey: Key.selectedAPIVersion.rawValue) }
     }
 
     /// The preferred API Version.
 
     public static var preferredAPIVersion: APIVersion? {
         get { return apiVersion(for: Key.preferredAPIVersion) }
-        set { storage.set(newValue?.rawValue, forKey: Key.preferredAPIVersion.rawValue) }
+        set { _storage.set(newValue?.rawValue, forKey: Key.preferredAPIVersion.rawValue) }
     }
 
     /// The domain of the backend to which the app is connected to.
 
     public static var domain: String? {
-        get { storage.string(forKey: Key.domain.rawValue) }
-        set { storage.set(newValue, forKey: Key.domain.rawValue) }
+        get { _storage.string(forKey: Key.domain.rawValue) }
+        set { _storage.set(newValue, forKey: Key.domain.rawValue) }
     }
 
     /// Whether the connected backend has federation enabled.
     ///
     /// If the backend has federation enabled, then it may be federating with other backends.
 
     public static var isFederationEnabled: Bool {
-        get { storage.bool(forKey: Key.isFederationEnabled.rawValue) }
-        set { storage.set(newValue, forKey: Key.isFederationEnabled.rawValue) }
+        get { _storage.bool(forKey: Key.isFederationEnabled.rawValue) }
+        set { _storage.set(newValue, forKey: Key.isFederationEnabled.rawValue) }
     }
 
     private static func apiVersion(for key: Key) -> APIVersion? {
         // Fetching an integer will default to 0 if no value exists for the key,
         // so explicitly check there is a value.
-        guard storage.object(forKey: key.rawValue) != nil else { return nil }
-        let storedValue = storage.integer(forKey: key.rawValue)
+        guard _storage.object(forKey: key.rawValue) != nil else { return nil }
+        let storedValue = _storage.integer(forKey: key.rawValue)
         return APIVersion(rawValue: Int32(storedValue))
     }
+}
+
+#if DEBUG
+
+// MARK: - Mock
+
+extension BackendInfo {
+    @_spi(MockBackendInfo)
+    public static func enableMocking() {
+        BackendInfo.mockStorage = MockUserDefaults(suiteName: UUID().uuidString)!
+    }
 
+    @_spi(MockBackendInfo)
+    public static func resetMocking() {
+        if let storage = mockStorage as? MockUserDefaults {
+            storage.removePersistentDomain(forName: storage.suiteName)
+        }
+        BackendInfo.mockStorage = nil
+    }
+}
+
+private class MockUserDefaults: UserDefaults {
```

Comment:
I created a subclass here just to have access to the `suiteName`. This allows us to delete it's data in `resetMocking()`